pkg/pcap/pcaps.go

```diff
@@ -0,0 +1,156 @@
+package pcap
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//
+// This is a big Pcaps struct holding caches for different types of pcap files
+// to be managed (pcap files per process, per containers and per commands). It
+// would be hard to keep all possible pcap files open forever (as tracee might
+// trace tons of processes, containers or commands).
+//
+// That is what the pcapCache interface is for. It can be implemented by any
+// different way of keeping pcap files and maintain those pcap files cached
+// while they're being used.
+//
+// At the end we have the pcap interface itself. It describes a pcap file being
+// kept opened on behalf of a process, a container or a command.
+//
+// NOTE: Pcaps is not thread safe, should be called from a single routine.
+// TODO: might replace cache_XXX and pcap_XXX files with generics.
+//
+
+//
+// All Pcap Implementations at Once
+//
+
+type Pcaps struct {
+	processes  pcapCache // ProcessPcap Cache (indexed by host_tid)
+	containers pcapCache // ContainersPcap Cache (indexed by container_id)
+	commands   pcapCache // CommandsPcap Cache (indexed by comm)
+}
+
+func NewPcaps(config int, output *os.File) (*Pcaps, error) {
+	var err error
+	//
+	// config:
+	//
+	// 1 (001): process:   1 pcap file per process (default)
+	// 2 (010): container: 1 pcap file per container
+	// 4 (011): command:   1 pcap file per command
+	//
+	// or a combination:
+	//
+	// 3 (011): process + container
+	// 5 (010): process + command
+	// 6 (110): container + command
+	// 7 (111): process + container + command
+	//
+
+	initializeGlobalVars(output)
+
+	var processes pcapCache
+	var containers pcapCache
+	var commands pcapCache
+
+	if config&0x1 == 0x1 {
```

Comment:
Then you could have `if config.captureProcesses` etc. which is much more immediately readable.

Reply:
I believe the const improved readability and is enough for now.